hooks/hooks.py

```diff
@@ -24,6 +25,10 @@
 
 @hooks.hook('config-changed')
 def config_changed():
+    if not version_check:
+        hookenv.log('CRITICAL', 'This charm requires juju 1.22.0 or greater.'
```

Comment:
I believe you are calling log() incorrectly here.  The message is first and the level is the second argument.

charmhelpers.core.hookenv.log(message, level=None)

    Write a message to the juju log

http://pythonhosted.org/charmhelpers/api/charmhelpers.core.hookenv.html#charmhelpers.core.hookenv.log